x/auth/ante/sigverify.go

```diff
@@ -218,17 +219,51 @@ func (sgcd SigGasConsumeDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simula
 // CONTRACT: Pubkeys are set in context for all signers before this decorator runs
 // CONTRACT: Tx must implement SigVerifiableTx interface
 type SigVerificationDecorator struct {
-	ak              AccountKeeper
-	signModeHandler *txsigning.HandlerMap
+	ak                   AccountKeeper
+	signModeHandler      *txsigning.HandlerMap
+	maxTxTimeoutDuration time.Duration
+	unorderedTxGasCost   uint64
+	unorderedTxManager   UnorderedNonceManager
```

Comment:
The only wonky thing is that we have these options that are not really signing related on the `signing` decorator

Reply:
i feel like an option controlling how far ahead a tx's nonce value is before its considered invalid is related enough to make sense here.

the gas cost does feel odd, but one could also consider its related to the cost of verifying a more complex tx type (unordered)